dist/log.py

```diff
@@ -6,390 +6,617 @@
+FieldType.Add('string',
+    'const char *', 'S', '\\"%s\\"', 'arg', [ '' ], False, False),
+FieldType.Add('WT_ITEM',
+    'WT_ITEM', 'u', '\\"%s\\"', '(char *)escaped->mem',
+    [ 'WT_ERR(__logrec_make_json_str(session, &escaped, &arg));',
+      'WT_ERR(__logrec_make_hex_str(session, &escaped, &arg));'], False, True),
+FieldType.Add('recno',
+    'uint64_t', 'r', '%" PRIu64 "', 'arg', [ '' ], False, False),
+FieldType.Add('uint32_t',
+    'uint32_t', 'I', '%" PRIu32 "', 'arg', [ '' ], False, False),
+# The fileid may have the high bit set. Print in both decimal and hex.
+FieldType.Add('uint32_id',
+    'uint32_t', 'I', [ '%" PRIu32 "', '\\"0x%" PRIx32 "\\"' ], 'arg', [ '', '' ], True, False),
+FieldType.Add('uint64_t',
+    'uint64_t', 'Q', '%" PRIu64 "', 'arg', [ '' ], False, False),
+
+
+class Field:
+    def __init__(self, field_tuple, field_idx, fields_count):
+        # Copy all attributes from FieldType object into this object.
+        self.__dict__.update(field_types[field_tuple[0]].__dict__)
+        self.fieldname = field_tuple[1]
+        self.fieldidx = field_idx
+        self.fieldcount = fields_count
+        self.is_last_field = (field_idx + 1 >= fields_count)
+
+        self.cintype = self.ctype + ('*' if self.byptr else '')
+        self.cindecl = self.ctype + (' *' if self.byptr else ' ') + self.fieldname
+        self.couttype = self.ctype + '*'
+        self.coutdecl = self.ctype + ' *' + self.fieldname + 'p'
+        self.clocaldef = self.ctype + ' ' + self.fieldname
+        self.n_setup = len(self.setup)
+        self.printf_arg = ' ' + self.printf_arg_templ.replace('arg', self.fieldname)
+
+        # Override functions for this type.
+        for func in ['pack_arg', 'unpack_arg',
+                     'struct_size_body', 'struct_pack_body', 'struct_unpack_body']:
+            if (func + '__' + self.typename) in dir(self):
+                setattr(self, func, getattr(self, func + '__' + self.typename))
+
+    def printf_fmt(self, ishex):
+        fmt = self.printf_fmt_templ
+        if type(fmt) is list:
+            fmt = fmt[ishex]
+        return fmt
+
+    def pack_arg(self):
+        return self.fieldname
+    # def pack_arg__WT_LSN(self):
+    #     return '%(name)s->l.file, %(name)s->l.offset' % {'name' : self.fieldname}
+
+    def unpack_arg(self):
+        return self.fieldname + 'p'
+    # def unpack_arg__WT_LSN(self):
+    #     return '&%(name)sp->l.file, &%(name)sp->l.offset' % {'name' : self.fieldname}
+
+    def printf_setup(self, i, nl_indent):
+        stmt = self.setup[i].replace('arg', self.fieldname)
+        return '' if stmt == '' else stmt + nl_indent
+
+    def struct_size_body(self):
+        return '__wt_vsize_uint(%s)' % self.fieldname
+    def struct_size_body__WT_LSN(self):
+        return '__wt_vsize_uint(%(name)s->l.file) + __wt_vsize_uint(%(name)s->l.offset)' \
+            % {'name' : self.fieldname}
+    def struct_size_body__WT_ITEM(self):
```

Comment:
Why do all of these have double underscores in the name? Please just use one underscore except at the beginning of the name.

Reply:
Double underscores are following the existing naming convention for similar functions. There is a ticket to fix it in the entire project: WT-11788.